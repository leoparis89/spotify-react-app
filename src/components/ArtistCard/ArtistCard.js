import React from 'react';
import PropTypes from 'prop-types';
import './ArtistCard.scss';
import {Link} from 'react-router-dom';
const ArtistCard = (props) => {

  const {name, image, popularity} = props.artist;
  return (

    <div className="col-md-3">
      <Link to={'artist/' + props.artist.id}>
        <div className="card">
          {image ?
            <img className="card-img-top vignette"
              style={{objectFit: 'cover'}}
              src={image} alt="Card image cap" />
            :
            <div className="vignette">
              <i className="fa fa-music" style={{fontSize: '200px'}}></i>
            </div>
          }
          <h3 className="card-body">{name}</h3>
          <p><span className="badge badge-pill badge-primary"> Popularity: {popularity}</span></p>
        </div>

      </Link>
    </div>
  );
};

ArtistCard.propTypes = {
  artist: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string,
  })
};

export default ArtistCard;
