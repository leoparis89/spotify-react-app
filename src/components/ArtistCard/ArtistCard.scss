.image{
  position:relative;
  overflow:hidden;
  padding-bottom:100%;
}
.image img{
  position: absolute;
  left: 0;
  width: 100%;
  height: 100%;
  object-fit: cover;
}
