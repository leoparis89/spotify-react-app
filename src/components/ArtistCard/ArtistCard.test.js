import React from 'react';
import ArtistCard from './ArtistCard';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router';


describe('Artist card', () => {
    
  const testProps = {
    artist: {
      id:'xyz',
      name:'metallica',
      image: 'htpp://foobar'
    }
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <ArtistCard {...testProps}/>
      </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
    
});
