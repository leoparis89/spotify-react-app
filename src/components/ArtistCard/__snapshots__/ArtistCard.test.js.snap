// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist card renders correctly 1`] = `
<div
  className="col-md-3"
>
  <a
    href="artist/xyz"
    onClick={[Function]}
  >
    <div
      className="card"
    >
      <div
        className="image"
      >
        <img
          alt="Card image cap"
          className=""
          src="htpp://foobar"
          style={
            Object {
              "objectFit": "cover",
            }
          }
        />
      </div>
      <h3
        className="card-body"
      >
        metallica
      </h3>
      <p>
        <span
          className="badge badge-pill badge-primary"
        >
           Popularity: 
        </span>
      </p>
    </div>
  </a>
</div>
`;
